README.md

```diff
@@ -6,41 +6,184 @@
 
-The standard practice in Java logging is to use fully-qualified name of each class as category;
-in our case, however, we're only ever using the categories provided above.
-FQCN (and method name), however, is always a part of the log message itself.
+When branch tree is created, Git Machete indicates relation between parent branch and each of its child branches.
+If edge between them is green that means child branch is in sync with its parent branch - in other words there are no commits in parent branch that don't belong to the child.
+But if there are some of that commits, then edge is red - we need to [rebase](#rebase) child branch onto the parent.
 
-By default, IntelliJ logs everything with level `INFO` and above into `idea.log` file. <br/>
-The exact location depends on a specific IntelliJ installation; check `Help` -> `Show Log in Files` to find out. <br/>
-Tip: use `tail -f` to watch the log file as it grows.
+Machete file editor will help you to manage `machete` file: it underlines any errors (bad indentation or nonexistent branches) and propose branch names based on local repository branches.
+When file editing is done, you can click button in top right corner of file editor to refresh machete branch tree.
 
-To enable logging in `DEBUG` level, add selected categories to list in `Help` -> `Diagnostic Tools` -> `Debug Log Settings`. <br/>
-A relatively small amount of `TRACE`-level logs is generated as well.
+![](docs/machete_file_editor.gif)
 
+For more information about `machete` file look at the [reference blog post](https://medium.com/virtuslab/make-your-way-through-the-git-rebase-jungle-with-git-machete-e2ed4dbacd02)
 
-## Development
+### Checking out branches
+
+With Git Machete you can easily check out branches that have been defined in `machete` file.
+Just right-click on the branch you wont to check out and select `Checkout Branch`.
+You can also double-click on the selected branch to check it out.
+Current branch is underlined in branch tree.
+
+![](docs/checkout.gif)
+
+### Listing commits
+
+Git Machete can display commits belonging to the branches.
+It displays only these commits that are unique for the selected branch (i.e. were never a tip of a different branch).
+When you click on eye button on the left side of plugin, you can toggle between show and hide commits.
+
+![](docs/toggle_listing_commits.gif)
+
+
+### Rebase
+
+Rebase with this plugin is easy!
+Just right-click on the child branch and from context menu select `(Checkout and) Rebase Branch onto Parent` and a standard IntelliJ dialog for interactive rebase will appear.
+
+![](docs/rebase.gif)
+
+### Push
+
+After a rebase, you might want to push rebased branch to remote.
+To do this, right-click on the branch you want to push and select `Push (Current) Branch` from context menu.
+Standard IntelliJ push dialog will appear but in case when we rebased selected branch (when branch we want to push diverge from its remote) only force push button will be available.
```

Comment:
I ain't true. This is the (un-standard) `Force Push Dialog`

edit
actually (IIRC) we use force push only if it's required XD

Reply:
\+ as a `force` may cause fear and pain to someone, maybe let's explain it shortly why the `force` is required :thinking:

Reply:
![image](https://user-images.githubusercontent.com/19799111/89276089-7c03d400-d643-11ea-9721-88fde2f41086.png)

Reply:
> I ain't true. This is the (un-standard) `Force Push Dialog`

Technically, yes but from user perspective it looks like standard push dialog (excepting forced action - force push or push)

Reply:
> actually (IIRC) we use force push only if it's required XD

Yes but basically it's when we rebased given branch. But ok, I will reword this to that way.

Reply:
> > I ain't true. This is the (un-standard) `Force Push Dialog`
> 
> Technically, yes but from user perspective it looks like standard push dialog (excepting forced action - force push or push)

It is easy to spot this lie. Let's be precise and make the users feel comfortable and safe (esp. in terms of `push --force`).

Reply:
So, I'd use just `Push dialog` wording (`Custom push dialog`, `Our push dialog` or other like that can sounds weird)